servers/rendering_server.cpp

```diff
@@ -402,9 +402,9 @@ Error RenderingServer::_surface_set_data(Array p_arrays, uint32_t p_format, uint
 				const Vector3 *src = array.ptr();
 				for (int i = 0; i < p_vertex_array_len; i++) {
 					Vector2 res = src[i].octahedron_encode();
-					int16_t vector[2] = {
-						(int16_t)CLAMP(res.x * 65535, 0, 65535),
-						(int16_t)CLAMP(res.y * 65535, 0, 65535),
+					uint16_t vector[2] = {
+						(uint16_t)CLAMP(res.x * 65535, 0, 65535),
+						(uint16_t)CLAMP(res.y * 65535, 0, 65535),
```

Comment:
The changes to this file should not be included in this PR as they are unrelated. 

Could please remove them from this PR, and open a new PR with only these changes (along with a link to the bug or description of the bug if it hasn't been reported yet).

Reply:
I'd like to push these changes to another Pull Request, but without them, the current pull request has an error in the automatic check [`Linux / Editor with doubles and GCC sanitizers`](https://github.com/godotengine/godot/actions/runs/4970081774/jobs/8893729737?pr=77047). I mentioned it in a previous copy of this pull request.

It appears because the engine is trying to use an image that did not match one of the possible types. In this case, the old code makes 3 attempts to initialize the image. One of these functions, despite the internal error, still initializes a small empty image. This masks the problem.

When using my image type detector, none of these functions are called and the image remains uninitialized. Then the value -1 is passed as the data length through the `int` type, which is cast to an unsigned type and becomes a very large number. This generates an error that fails the check I mentioned above.


https://github.com/godotengine/godot/pull/77047#issuecomment-1546793960

Reply:
I don't know how that can be as this PR only changes image/texture functionality and these lines of code are only run when creating a mesh surface. They have nothing to do with images/textures

Reply:
Without these lines of code, the check that I described above will not pass. Correct reading of the image data leads to the discovering of a hidden problem.

As a result, the code works as it should - the wrong image is really not loaded. The image object remains uninitialized (that's right).

As a result of this, here, in the `RenderingServer`, the `x` and `y` components get value 1. This results in a large number (65535) in the expression (it is OK), and a negative number after the wrong cast (it is not OK). This causes Godot to fail on this check.

Therefore, I cannot remove these changes from my PR. And we need to find and replace the image that Godot is trying to load.